### PR TITLE
[compact-interactive-list] Initial component setup

### PR DESCRIPTION
**Pull request by adoroshk**
### Summary
This PR sets up the base terra-compact-interactive-list component to prepare for development.

These changes are based on the following PRs:

- Terra File Path changes from https://github.com/cerner/terra-framework/pull/1771.
- Terra Folder Tree https://github.com/cerner/terra-framework/pull/1855

### Testing

The initial WDIO and Jest test were set up, but they don't have actual tests yet and will be updated with added functionality.
This change was tested using:

- [ ] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [x] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR is a part of:**

UXPLATFORM-9557

### CODE REVIEW

**Comment by adoroshk** [comment]
The terra-table changes are only for fixing broken links to the terra-framework repo in terra-table package.json, and I think they don't need a changelog entry, but correct me if I am wrong

File: packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+   * String that labels the table for accessibility. If ariaLabelledBy is specified, ariaLabel will not be used.
+   */
+  ariaLabel: PropTypes.string,
+
+  /**
+   * Boolean indicating whether or not the table columns should be displayed. Setting the value to false will hide the columns,
+   * but the voice reader will use the column header values for a11y.
+   */
+  hasColumnHeaders: PropTypes.bool,
+};
```
Comment (sdadn) [inline packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx:32]: is this prop correct?

File: packages/terra-compact-interactive-list/src/proptypes/cellShape.js
Comment (sdadn) [inline packages/terra-compact-interactive-list/src/proptypes/cellShape.js:1]: Do we need these shapes?

File: packages/terra-compact-interactive-list/src/utils/constants.js
[the remark this answers is not in the document]
Comment (adoroshk) [reply packages/terra-compact-interactive-list/src/utils/constants.js:1]: we will need them in the future, but you are right, no need to have then at that point. Removed: https://github.com/cerner/terra-framework/pull/1861/commits/4392403f5207d7912ba8948206ef39898f607aad

File: packages/terra-compact-interactive-list/src/proptypes/validators.js
[the remark this answers is not in the document]
Comment (adoroshk) [reply packages/terra-compact-interactive-list/src/proptypes/validators.js:1]: same as above, removed in https://github.com/cerner/terra-framework/pull/1861/commits/4392403f5207d7912ba8948206ef39898f607aad

File: packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx (hunk 1 above)
Comment (adoroshk) [reply packages/terra-compact-interactive-list/src/CompactInteractiveList.jsx]: I edited the prop to default to false and edited description per that. Currently the plan is to not display the headers, but allow users to turn them on if they need them.

File: packages/terra-compact-interactive-list/src/proptypes/cellShape.js
Comment (adoroshk) [reply packages/terra-compact-interactive-list/src/proptypes/cellShape.js:1]: I guess we need the shapes. I edited them a bit though: https://github.com/cerner/terra-framework/pull/1861/commits/4392403f5207d7912ba8948206ef39898f607aad

**Review by sycombs — APPROVED** [review]

**Review by artpark — APPROVED** [review]

**Pull request merged**